nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -45,65 +45,153 @@ def _setup_tokenizer(self, tokenizer_cfg: DictConfig):
+            self.tokenizer_cfg.pop('type')
+            self.tokenizer_cfg.pop('hf_kwargs', {})
+
+    def _make_tokenizer(self, tokenizer_cfg: DictConfig, lang=None):
+
+        tokenizer_type = tokenizer_cfg.get('type')
+        tokenizer_dir = tokenizer_cfg.get('dir')
+
+        if tokenizer_type not in ['bpe', 'wpe']:
+            raise ValueError(
+                '`tokenizer.type` must be either `bpe` for SentencePiece tokenizer or' '`wpe` for BERT based tokenizer'
+            )
+
+        # defaults
+        model_path = None
+        vocab_path = None
+        spe_vocab_path = None
+
+        if tokenizer_type == 'bpe':
             # This is a BPE Tokenizer
-            if 'model_path' in self.tokenizer_cfg:
-                model_path = self.tokenizer_cfg.get('model_path')
+            if 'model_path' in tokenizer_cfg:
+                model_path = tokenizer_cfg.get('model_path')
+            else:
+                model_path = os.path.join(tokenizer_dir, 'tokenizer.model')
+            if lang is not None:
+                model_path = self.register_artifact('tokenizer.tokenizers.' + lang + '.model_path', model_path)
             else:
-                model_path = os.path.join(self.tokenizer_dir, 'tokenizer.model')
-            model_path = self.register_artifact('tokenizer.model_path', model_path)
-            self.model_path = model_path
+                model_path = self.register_artifact('tokenizer.model_path', model_path)
 
-            if 'special_tokens' in self.tokenizer_cfg:
-                special_tokens = self.tokenizer_cfg['special_tokens']
+            if 'special_tokens' in tokenizer_cfg:
+                special_tokens = tokenizer_cfg['special_tokens']
 
                 if special_tokens is not None:
-                    raise ValueError("`special_tokens` are no longer supported for SentencePiece based tokenizers.")
+                    raise ValueError('`special_tokens` are no longer supported for SentencePiece based tokenizers.')
 
             # Update special tokens
-            self.tokenizer = tokenizers.SentencePieceTokenizer(model_path=model_path)
+            tokenizer = tokenizers.SentencePieceTokenizer(model_path=model_path)
+
+            if 'vocab_path' in tokenizer_cfg:
+                vocab_path = tokenizer_cfg.get('vocab_path')
+            else:
+                vocab_path = os.path.join(tokenizer_dir, 'vocab.txt')
 
-            if 'vocab_path' in self.tokenizer_cfg:
-                vocab_path = self.tokenizer_cfg.get('vocab_path')
+            if lang is not None:
+                vocab_path = self.register_artifact('tokenizer.tokenizers.' + lang + '.vocab_path', vocab_path)
             else:
-                vocab_path = os.path.join(self.tokenizer_dir, 'vocab.txt')
-            vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
-            self.vocab_path = vocab_path
+                vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
```

Comment:
Missed self.vocab_path assignment.

Reply:
can't make the assignment because there is more than one potentially; this method just builds, does not assign